pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -253,6 +253,10 @@ public UnAckedMessageTracker getUnAckedMessageTracker() {
 
     @Override
     public CompletableFuture<Void> unsubscribeAsync() {
+        if (!shouldTearDown()) {
```

Comment:
I don't think we need this for `unsubscribe`.

technically if any consumer in a subscription unsubscribe, the subscription is unsubscribed.

Reply:
Great! I was dubious about `unsubscribe`.
Thanks, removed! :+1: